File: remote/src/main/java/com/kelsos/mbrc/annotations/Shuffle.java
package com.kelsos.mbrc.annotations;

import android.support.annotation.StringDef;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

public class Shuffle {
  public static final String OFF = "off";
  public static final String AUTODJ = "autodj";
  public static final String ON = "on";
  public static final String TOGGLE = "";
  public static final String UNDEF = "undef";

  private Shuffle() {
    //no instance
  }

  @StringDef({
      OFF,
      AUTODJ,
      ON,
      TOGGLE,
      UNDEF
  })

  @Retention(RetentionPolicy.SOURCE)
  public @interface State {
  }
}

File: app/src/main/kotlin/com/kelsos/mbrc/annotations/Shuffle.kt
package com.kelsos.mbrc.annotations

import android.support.annotation.StringDef

object Shuffle {
    const val OFF = "off"
    const val AUTODJ = "autodj"
    const val ON = "on"
    const val TOGGLE = ""
    const val UNDEF = "undef"

    @StringDef(OFF, AUTODJ, ON, TOGGLE, UNDEF)

    @kotlin.annotation.Retention(AnnotationRetention.SOURCE)
    annotation class State
}
